application/controllers/admin/Tokens.php

```diff
@@ -1417,10 +1417,11 @@ public function email(int $iSurveyId)
                     $mail->setFrom(Yii::app()->request->getPost('from_' . $emrow['language']));
                     $mail->rawSubject = $sSubject[$emrow['language']];
                     $mail->rawBody = $sMessage[$emrow['language']];
-                    if (!App()->request->getPost('bypassdatecontrol') == '1' && trim($emrow['validfrom']) != '' && convertDateTimeFormat($emrow['validfrom'], 'Y-m-d H:i:s', 'U') * 1 > date('U') * 1) {
+                    $now = dateShift(date("Y-m-d H:i:s"), "Y-m-d H:i", Yii::app()->getConfig('timeadjust'));
+                    if (!App()->request->getPost('bypassdatecontrol') == '1' && trim($emrow['validfrom']) != '' && strtotime($emrow['validfrom']) > strtotime($now)) {
```

Comment:
`!App()->request->getPost('bypassdatecontrol') == '1'` ???

Better : `!App()->request->getPost('bypassdatecontrol') `or `App()->request->getPost('bypassdatecontrol') != '1'` 

Else can we wrap all in a if : 
```
if (!App()->request->getPost('bypassdatecontrol')) {
    if (trim($emrow['validfrom']) != '' && strtotime($emrow['validfrom']) > strtotime($now)) {
        // action
    } elseif (trim($emrow['validuntil']) != '' && strtotime($emrow['validuntil']) < strtotime($now) {
        // action
    }
}
```

really more understandable